hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/prefetch/TestS3ARemoteInputStream.java

```diff
@@ -121,8 +121,8 @@ public void testRead() throws Exception {
 
   private void testReadHelper(S3ARemoteInputStream inputStream, int bufferSize)
       throws Exception {
-    assertEquals(bufferSize, inputStream.available());
     assertEquals(0, inputStream.read());
+    assertEquals(bufferSize - 1, inputStream.available());
```

Comment:
can you add string of what is being tested. the assertEquals prints the values, but it is important to know what is being validated. same below.

actually, given the #of calls, what about a method `assertAvailable(InputStream, int)` to do the probe everywhere